common/flatpak-utils.c

```diff
@@ -4980,6 +4980,14 @@ flatpak_repo_update (OstreeRepo   *repo,
                                                     g_variant_get_data (old_index),
                                                     g_variant_get_size (old_index));
 
+  /* Release the memory-mapped summary index file before replacing it,
+     to avoid failure on filesystems like cifs */
+  if (old_index)
+    {
+      g_variant_unref (old_index);
+      old_index = NULL;
+    }
```

Comment:
```suggestion
  g_clear_pointer (&old_index, g_variant_unref);
```

Reply:
Thanks for pointing out g_clear_pointer(). Updated and squashed.